docs/client/rest.md

```diff
@@ -0,0 +1,770 @@
+
+```go title="Signature"
+func (c *Client) SetPathParamsWithStruct(v any) *Client
+```
+
+## DelPathParams
+
+DelPathParams method deletes single or multiple path params field and its values in client.
+
+```go title="Signature"
+func (c *Client) DelPathParams(key ...string) *Client
+```
+
+## Cookie
+
+Cookie returns the cookie be set in request instance.
+If cookie doesn't exist, return empty string.
+
+```go title="Signature"
+func (c *Client) Cookie(key string) string
+```
+
+## SetCookie
+
+SetCookie method sets a single cookie field and its value in the client instance.
+These cookies will be applied to all requests raised from this client instance.
+Also it can be overridden at request level cookie options.
+
+```go title="Signature"
+func (c *Client) SetCookie(key, val string) *Client
+```
+
+## SetCookies
+
+SetCookies method sets multiple cookies field and its values at one go in the client instance.
+These cookies will be applied to all requests raised from this client instance. 
+Also it can be overridden at request level cookie options.
+
+```go title="Signature"
+func (c *Client) SetCookies(m map[string]string) *Client
+```
+
+## SetCookiesWithStruct
+
+SetCookiesWithStruct method sets multiple cookies field and its values at one go in the client instance.
+These cookies will be applied to all requests raised from this client instance.
+Also it can be overridden at request level cookies options.
+
+```go title="Signature"
+func (c *Client) SetCookiesWithStruct(v any) *Client
+```
+
+## SetCookiesWithStruct
+
+SetCookiesWithStruct method sets multiple cookies field and its values at one go in the client instance.
+These cookies will be applied to all requests raised from this client instance.
+Also it can be overridden at request level cookies options.
+
+```go title="Signature"
+func (c *Client) SetCookiesWithStruct(v any) *Client
+```
+
+## DelCookies
+
+DelCookies method deletes single or multiple cookies field and its values in client.
+
+```go title="Signature"
+func (c *Client) DelCookies(key ...string) *Client
+```
+
+## SetTimeout
+
+SetTimeout method sets timeout val in client instance.
+This value will be applied to all requests raised from this client instance.
+Also, it can be overridden at request level timeout options.
+
+```go title="Signature"
+func (c *Client) SetTimeout(t time.Duration) *Client
+```
+
+## Debug
+
+Debug enable log debug level output.
+
+```go title="Signature"
+func (c *Client) Debug() *Client
+```
+
+## DisableDebug
+
+DisableDebug disables log debug level output.
+
+```go title="Signature"
+func (c *Client) DisableDebug() *Client
+```
+
+## SetCookieJar
+
+DisableDebug disables log debug level output.
+
+```go title="Signature"
+func (c *Client) SetCookieJar(cookieJar *CookieJar) *Client
+```
+
+## SetCookieJar
+
+SetCookieJar sets cookie jar in client instance.
+
+```go title="Signature"
+func (c *Client) SetCookieJar(cookieJar *CookieJar) *Client
+```
+
+## Get
```

Comment:
put these REST methods at the top in a separate section and the configuration methods in a separate section